test/e2e/framework/conformance/conformance.go

```diff
@@ -0,0 +1,986 @@
+	}
+
+	// Set up informers, optionally also in the namespace.
+	// After each step we check that the informer catch up.
+	// They get stopped through test context cancellation.
+	resourceInformer := dynamicinformer.NewFilteredDynamicInformer(f.DynamicClient, gvr, "", 0, nil, func(options *metav1.ListOptions) {
+		options.LabelSelector = listOptions.LabelSelector
+	})
+	listResource := func(_ context.Context) ([]runtime.Object, error) {
+		return resourceInformer.Lister().List(labels.Everything())
+	}
+	go resourceInformer.Informer().RunWithContext(ctx)
+	informersHaveSynced := []cache.InformerSynced{resourceInformer.Informer().HasSynced}
+	var namespaceInformer informers.GenericInformer
+	var listNamespace func(_ context.Context) ([]runtime.Object, error)
+	if tc.IsNamespaced() {
+		namespaceInformer = dynamicinformer.NewFilteredDynamicInformer(f.DynamicClient, gvr, f.Namespace.Name, 0, nil, func(options *metav1.ListOptions) {
+			options.LabelSelector = listOptions.LabelSelector
+		})
+		listNamespace = func(_ context.Context) ([]runtime.Object, error) {
+			return namespaceInformer.Lister().List(labels.Everything())
+		}
+		go namespaceInformer.Informer().RunWithContext(ctx)
+		informersHaveSynced = append(informersHaveSynced, namespaceInformer.Informer().HasSynced)
+	}
+	if !cache.WaitForNamedCacheSyncWithContext(ctx, informersHaveSynced...) {
+		ginkgo.Fail("informers should have synced and didn't")
+	}
+
+	// matchObject generates a matcher which checks the result of a list operation
+	// against the expected object. Content verification is optional. Without it,
+	// only the namespace and name are checked.
+	matchObject := func(expectedObject *unstructured.Unstructured) gtypes.GomegaMatcher {
+		return &matchObjectList{expectedObject: expectedObject, verifyContent: tc.VerifyContent()}
+	}
+	gomega.Expect(listResource(ctx)).Should(matchObject(nil), "initial list of resources from informer cache")
+	if listNamespace != nil {
+		gomega.Expect(listNamespace(ctx)).Should(matchObject(nil), "initial list of namespace from informer cache")
+	}
+
+	// Verification of the result after each step (= what)
+	// can check the content or be limited to just name/namespace.
+	//
+	// Also checks the informers. This is a bit redundant after read-only steps,
+	// but then it's also fast and thus can be done more often than strictly necessary.
+	verify := func(what string, expected, actual *unstructured.Unstructured) {
+		ginkgo.GinkgoHelper()
+
+		if tc.VerifyContent() {
+			diff := compareObjects(expected, actual)
+			if diff != "" {
+				framework.Failf("%s: unexpected actual object (- expected, + actual):\n%s", what, diff)
+			}
+		} else {
+			gomega.Expect(actual.GetName()).Should(gomega.Equal(expected.GetName()), "%s: name in returned object", what)
+			gomega.Expect(actual.GetNamespace()).Should(gomega.Equal(expected.GetNamespace()), "%s: namespace in returned object", what)
+		}
```

Comment:
Add a check that the resourceVersion invariant from KEP-5504 is valid

```
gomega.Expect(actual).To(apimachineryutils.HaveValidResourceVersion())
```

This will ensure all future conformance-tested APIs continue to have resourceVersions that match documented behavior for Kubernetes APIs

Reply:
Saw this too late, not included in latest update yet. Will do next...

Reply:
Added.